pkg/osm/utils_test.go

```diff
@@ -0,0 +1,160 @@
+package osm
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+func TestGetOSMControllerDeployment(t *testing.T) {
+	tests := []struct {
+		name        string
+		deployments []*v1.Deployment
+		namespace   string
+		pass        bool
+	}{
+		{
+			name: "there are no deployments in the controller namespace",
+			deployments: []*v1.Deployment{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notosmcontrollerdeployment",
+						Namespace: "notosmnamespace",
+						Labels: map[string]string{
+							constants.OSMAppInstanceLabelKey: "osm",
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+			},
+			namespace: "osmnamespace",
+			pass:      false,
+		},
+		{
+			name: "multiple deployments in the controller namespace",
+			deployments: []*v1.Deployment{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osmcontrollerdeployment",
+						Namespace: "osmnamespace",
+						Labels: map[string]string{
+							constants.OSMAppInstanceLabelKey: "osm",
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osmcontrollerdeployment2",
+						Namespace: "osmnamespace",
+						Labels: map[string]string{
+							constants.OSMAppInstanceLabelKey: "osm",
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+			},
+			namespace: "osmnamespace",
+			pass:      false,
+		},
+		{
+			name: "there is one deployment in the controller namespace",
+			deployments: []*v1.Deployment{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "osmcontrollerdeployment",
+						Namespace: "osmnamespace",
+						Labels: map[string]string{
+							constants.OSMAppInstanceLabelKey: "osm",
+							constants.OSMAppVersionLabelKey:  "v0.9.1",
+							"app":                            constants.OSMControllerName,
+						},
+					},
+				},
+			},
+			namespace: "osmnamespace",
+			pass:      true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			assert := tassert.New(t)
+			objs := make([]runtime.Object, len(test.deployments))
+			for i := range test.deployments {
+				objs[i] = test.deployments[i]
+			}
+			k8s := fake.NewSimpleClientset(objs...)
+			deployment, err := GetOSMControllerDeployment(k8s, test.namespace)
+			if test.pass {
```

Comment:
I think we should change the `pass` field to `expErr` so that we can have the check as 			
`assert.Equal(test.expErr, err != nil)` and don't have to use an if statement